r_code/panel_estimators.R

```diff
@@ -0,0 +1,750 @@
+      )
+  }
+  rownames(causalimpact_series_output) <- NULL
+  # Reorder columns for consistency across methods.
+  causalimpact_series_output <- causalimpact_series_output %>%
+    dplyr::select(
+      tidyselect::all_of(c(
+        time_var, id_var, "response",
+        "point.pred", "point.effect",
+        "pct.effect", "Treatment_Period"
+      )),
+      dplyr::everything()
+    )
+  return(causalimpact_series_output)
+}
+
+#' Estimate the gsynth point estimates of the outcome.
+#'
+#' @inheritParams EstimateCausalImpact
+#' @param ... Additional parameters to pass to gsynth::gsynth function. These
+#'    include the estimator (="mc", "ife"), X variables, and se (among others).
+#'
+#' @return A long form tibble with the gsynth method estimates of the
+#'    outcome variable (missing outcome imputation), as well as point and
+#'    percent effects.
+EstimateGsynthSeries <- function(data_full,
+                                 id_var = "entry",
+                                 time_var = "period",
+                                 treat_indicator = "treatperiod_0",
+                                 outcome_var = "target",
+                                 counterfac_var = "counter_factual", ...) {
+  unnamed_args <- list(...)
+  se <- ifelse(is.null(unnamed_args$se), F, unnamed_args$se)
+  inference <- ifelse(is.null(unnamed_args$inference),
+    "nonparametric", unnamed_args$inference
+  )
+  # Need to turn parallel off as it's on by default.
+  parallel <- ifelse(is.null(unnamed_args$parallel), F, unnamed_args$parallel)
+  # Call the gsynth function on the given input.
+  gsynth_agg_te_all_t <- gsynth::gsynth(
+    Y = outcome_var, D = treat_indicator, data = data_full,
+    index = c(id_var, time_var), parallel = parallel, ...
+  )
+  # Determine whether any uncertainty parameters were called.
+  # If user specified se and inference output, provide it.
+  if (se & inference == "parametric") {
+    # Get all individual effects for all time periods.
+    # Renames the multi-dimensional array output for easy pivotting.
+    gsynth_indiv_te_series <-
+      gsynth_agg_te_all_t$est.ind[, , 1:(gsynth_agg_te_all_t$Ntr)] %>%
+      tibble::as_tibble() %>%
+      dplyr::mutate(!!as.name(time_var) := seq_len(nrow(.))) %>%
+      janitor::clean_names() %>%
+      stats::setNames(ifelse(stringr::str_count(names(.), "_") > 1,
+        stringr::str_replace(names(.), "[_]", ""),
+        names(.)
+      ))
+    # Pivot the wide (one row per time) tibble of effects (and CI) per entry
+    # into long data, where each period-entry combination has a column for
+    # effect, CI.
+    gsynth_series_output <- gsynth_indiv_te_series %>%
+      tidyr::pivot_longer(
+        -!!as.name(time_var),
+        names_to = c(".value", id_var),
+        names_sep = "_"
+      ) %>%
+      dplyr::mutate(!!as.name(id_var) := as.numeric(!!as.name(id_var))) %>%
+      dplyr::rename(point.effect = eff)
+  }
+  else {
```

Comment:
I'm fairly confident that's the case. When nonparametric (the only other "inference" option) is specified, the above method is null. I believe when standard errors are computed parametrically the output is store in a unique way, and so that if statement catches that.